internal/controller/ols_app_redis_assets.go

```diff
@@ -70,28 +132,63 @@ func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig)
 	return &deployment, nil
 }
 
-func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*corev1.Service, error) {
-	DeploymentSelectorLabels := map[string]string{
-		"app.kubernetes.io/component":  "redis-server",
-		"app.kubernetes.io/managed-by": "lightspeed-operator",
-		"app.kubernetes.io/name":       "lightspeed-service-redis",
-		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+// updateRedisDeployment updates the deployment based on CustomResource configuration.
+func (r *OLSConfigReconciler) updateRedisDeployment(ctx context.Context, existingDeployment, desiredDeployment *appsv1.Deployment) error {
+	changed := false
+
+	// Validate deployment annotations.
+	if existingDeployment.Annotations == nil ||
+		existingDeployment.Annotations[RedisConfigHashKey] != r.stateCache[RedisConfigHashStateCacheKey] {
```

Comment:
@vishnuchalla Do we really need 2 stateCaches ? One for OLS App Server one for Redis ? I am thinking we have ONE CR instance .  If any part on the CR changes, we can Sync all objects . Please help me understand the reason here.

Reply:
the state cache is for the configmap, if the config map changes, we trigger the rollout of deployment by modifying the annotation in its container template. so it is good to have one for the application server and one for the cache server as they both have a dedicated configmap.

Reply:
Ahh too fast, @raptorsun got me covered. Thank you.